### DISCUSSION TITLE
Mutate cache data

### DISCUSSION THREAD

**Discussion by bartoszkrawczyk2** [solidjs/solid-start#1225] [Q&A]
Hi!

Is there a pattern to optimistic mutate data loaded with `cache` and `createAsync` ? Similar to `mutate` in `createResource`.

Would be great to be able to preload data on the server with `use server` in `cache`, but be able to update it on client side while calling `revalidate` to show optimistic state.

Thanks!

**Comment by ryansolid** [comment]
There is a `cache.set` method. We need to document it though. It takes a key, value.

**Reply by maximusdionyssopoulos** [reply]
Hi!

Does this update the value of the data from the createAsync?

For example:

```
const loadAboutData = cache(async () => {
  return new Promise((resolve) =>
    setTimeout(resolve, 500, "Solid")
  ) as Promise<string>;
}, "about");


export default function About() {
  const name = createAsync(() => loadAboutData());
  return (
    <section class="p-8">
      <Suspense fallback={<p>...</p>}>
        <p>{name()}</p>
      </Suspense>
      <button
        onClick={() => {
          // optimistically update the data

cache.set(loadAboutData.key, "TEST");
// mutation
        }}>
        Test
      </button>
    </section>
  );
}
```

Would this be able to change name() from "Solid" to "TEST" or is there another way to do this?

Thanks!
